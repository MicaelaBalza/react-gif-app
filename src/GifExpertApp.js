import { useState } from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2 className="page-title">GIF APP</h2>
      <div className="page-content">
        <AddCategory setCategories={setCategories} />
        {categories.map(cat => <GifGrid category={cat} key={cat} />)}
      </div>
    </>
  )
};

export default GifExpertApp;
