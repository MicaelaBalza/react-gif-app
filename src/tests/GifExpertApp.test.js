import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas GifExpertApp', () => {
  test('debe renderizar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar una lista de categorÃ­as', () => {
    const categories = ['Pokemon', 'Digimon'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
})
