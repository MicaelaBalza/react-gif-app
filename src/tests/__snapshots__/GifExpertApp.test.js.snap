// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h2
    className="page-title"
  >
    GIF APP
  </h2>
  <div
    className="page-content"
  >
    <AddCategory
      setCategories={[Function]}
    />
  </div>
</Fragment>
`;
